store/cache.go

```diff
@@ -143,11 +144,42 @@ func (cm *CacheManager) cleanup() error {
 		return nil
 	}
 
-	sort.Sort(changesByReverseMtime(fil))
-	for _, fi := range fil[cm.maxItems:] {
-		if err := os.Remove(cm.path(fi.Name())); err != nil && os.IsNotExist(err) {
+	var lastErr error
+	sort.Sort(changesByMtime(fil))
+	deleted := 0
+	for _, fi := range fil {
+		path := cm.path(fi.Name())
+		n, err := hardLinkCount(path)
+		if err != nil {
+			lastErr = err
+			continue
+		}
+		// If the file is referenced in the filesystem somewhere
+		// else our copy is "free" so skip it.
+		if n > 1 {
+			continue
+		}
+		if err := os.Remove(path); err != nil && os.IsNotExist(err) {
 			return err
```

Comment:
Shouldn't we try to process remaining files despite of failure of a single file?

Reply:
Yes, good point.